project/JavalibIRCleaner.scala

```diff
@@ -161,17 +157,19 @@ final class JavalibIRCleaner(baseDirectoryURI: URI) {
     }
   }
 
-  private def getJSTypes(trees: Iterator[ClassDef]): Map[ClassName, ClassDef] =
-    trees.filter(_.kind.isJSType).map(t => t.className -> t).toMap
+  private def getJSTypes(trees: Stream[ClassDef]): JSTypes = {
+    trees.filter(_.kind.isJSType).reduce[JSTypes](
+        Map.empty, (m, v) => m.updated(v.className, v.jsNativeLoadSpec), _ ++ _)
```

Comment:
The reduction here is a bit of a mouthful. Alternatives:
- Write a Collector for (immutable) Scala maps.
- Switch to Java collections, use `Collectors.toMap` (was my original plan, but Java's usage site variance makes Scala type inference very unhappy, so it's not super readable).

Reply:
The `reduce` doesn't look too bad to me. `Collectors.toMap` would have been nice, but it doesn't make things more readable, it's not worth it.

Writing a `Collector` seems quite a bit overkill to save readability of 2 lines of code. 😅